frontend/packages/aquascope-editor/src/lib.ts

```diff
@@ -195,13 +132,38 @@ let parseWithDelimiters = (
   return { type: "ok", code: outputCode.join(""), ranges };
 };
 
+let buildStepperConfig = (config: StepperConfig) => {
+  config.focusedCharPos = [];
+  config.focusedPaths = new Map();
+  return (toks: string[], range: Range): boolean => {
+    let hasTag = toks.length >= 5 && toks.slice(0, 5).join("") === "step:";
+    return (
+      !hasTag ||
+      (() => {
+        let remaining = toks.slice(5).join("");
+        let commaSep = remaining.split(",", 2).map(s => s.trim());
+        if (commaSep.includes("focus")) {
+          config.focusedCharPos!.push(range.char_start);
+          commaSep.splice(commaSep.indexOf("focus"), 1);
+        }
+        if (commaSep[0]) {
+          config.focusedPaths!.set(range.char_start, commaSep[0]);
+        }
+        return false;
+      })()
+    );
+  };
+};
+
 export class Editor {
   private view: EditorView;
   private interpreterContainer: HTMLDivElement;
+  private editorContainer: HTMLDivElement;
   private markedRanges: Range[];
```

Comment:
Let's coalesce all the configuration generated by annotations into a single data structure (i.e. put `markedRanges` and `focusStepConfigs` into a struct.)